src/main/java/org/example/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.example.bookstore.repository.impl;
+
+import jakarta.persistence.criteria.CriteriaQuery;
+import java.util.List;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book: " + book, e);
```

Comment:
Better use DataProcessingException (but you remove this method during this module, so you can leave it)